outboundSender.go

```diff
@@ -368,6 +387,28 @@ func (obs *CaduceusOutboundSender) Queue(msg *wrp.Message) {
 		return
 	}
 
+	//check the partnerIDs
+	if len(obs.listener.PartnerIDs) == 0 {
+		switch obs.noPIDAction {
+		case "custom":
+			obs.listener.PartnerIDs = obs.customPIDs
+		case "donotsend":
+			return
+		}
+
+	} else {
+		hasPID := false
```

Comment:
Consider moving this logic to a helper function, probably an unexported function.  That makes this logic easier to unit test, and reduces the code noise of the enclosing function.

Alternatively, you could make this a method of whatever `obs.listener` is.